R/cnd-handlers.R

```diff
@@ -214,12 +216,19 @@ handler_call <- quote(function(cnd) {
     } else {
       except <- ""
     }
+    inherit <- .subset2(.subset2(cnd, "rlang"), "inherit")
   }
+
   while (!is_null(cnd)) {
-    if (inherits(cnd, CLASS) && !inherits(cnd, except)) {
+    if (inherits(cnd, CLASS)) {
       out <- handlers[[I]](cnd)
       if (!inherits(out, "rlang_zap")) throw(out)
     }
+
+    if (is_false(inherit)) {
+      return()
+    }
+
     cnd <- .subset2(cnd, "parent")
```

Comment:
Should `inherit` be updated after each iteration to the `inherit` slot of next parent?

Reply:
oh good point